src/hooks/useIntegratorEventHandlers.ts

```diff
@@ -0,0 +1,37 @@
+import { useAtom } from 'jotai'
+import { useEffect } from 'react'
+import { onReviewSwapClickAtom, onTokenSelectorClickAtom } from 'state/swap'
+import { onConnectWalletClickAtom } from 'state/wallet'
+
+interface UseIntegratorEventHandlersArgs {
+  onConnectWalletClick?: () => void | Promise<boolean>
+  onReviewSwapClick?: () => void | Promise<boolean>
+  onTokenSelectorClick?: () => void | Promise<boolean>
+}
+
+export default function useIntegratorEventHandlers(handlers: UseIntegratorEventHandlersArgs): void {
+  const [onReviewSwapClick, setOnReviewSwapClick] = useAtom(onReviewSwapClickAtom)
```

Comment:
I think it's overkill to check against the old value of `onReviewSwapClick`: it adds a dependency, and if this behaves like useState, then setting the current value should be a nop anyways.

Can all of these handlers be simplified to

    const { onReviewSwapClick } = props
    const setOnReviewSwapClick = useSetAtom(onReviewSwapClickAtom)
    useEffect(() => setOnReviewSwapClick(onReviewSwapClick), [onReviewSwapClick])

Reply:
@kristiehuang

Reply:
when I tried this without the check I'd get 'Too many re-renderers' errors

Reply:
hm - I'd get 'too many re-renders' errors without the check but can't reproduce it now. might have fixed something in the meantime, seems fair to me to simplify if it works